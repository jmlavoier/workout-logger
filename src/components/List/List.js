import React from 'react';
import PropTypes from 'prop-types';

import Container from 'components/Container';
import Box from 'components/Box';
import OrderButton from 'components/OrderButton';

import ItemList from '../ItemList';
import styles from './List.sass';


const handleClickRemove = clickRemove => id => () => {
  clickRemove(id);
};

const List = ({
  items,
  changeOrder,
  clickRemove,
  orderBy,
}) => (
  <table className={styles.list} >
    <thead>
      <tr>
        <td>
          <OrderButton changeOrder={changeOrder} fieldName="timeSpent" orderBy={orderBy}>
            Tempo
          </OrderButton>
        </td>
        <td>
          <OrderButton changeOrder={changeOrder} fieldName="workoutType" orderBy={orderBy}>
            Tipo
          </OrderButton>
        </td>
        <td>
          <OrderButton changeOrder={changeOrder} fieldName="date" orderBy={orderBy}>
            Data
          </OrderButton>
        </td>
        <td>
          &nbsp;
        </td>
      </tr>
    </thead>
    <tbody>
      {items.length > 0 ? (
        items.map(item => <ItemList key={item.id} item={item} onClickRemove={handleClickRemove(clickRemove)} />)
      ) : (
        <tr>
          <td colSpan="4">
            <Container>
              <Box>Lista de workouts est√° vazia...</Box>
            </Container>
          </td>
        </tr>
      )}
    </tbody>
  </table>
);

List.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  changeOrder: PropTypes.func.isRequired,
  clickRemove: PropTypes.func.isRequired,
  orderBy: PropTypes.shape({
    field: PropTypes.string,
    asc: PropTypes.bool,
  }).isRequired,
};

List.defaultProps = {
  items: [],
};

export default List;
