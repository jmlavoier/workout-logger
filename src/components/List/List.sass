@import 'css/variables'

.list
  width: 100%

  thead
    tr
      td
        background-color: $color-gray-1
        padding: 5px

      td:nth-child(1)
        width: 100px

      td:nth-child(2)
        width: auto

      td:nth-child(3)
        width: 100px

      td:nth-child(4)
        width: 40px
