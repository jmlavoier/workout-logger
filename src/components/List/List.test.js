import React from 'react';
import { shallow, mount } from 'enzyme';

import List from './List';

const mockList = [
  { 
    id: 0,
    timeSpent: '1:00',
    workoutType: 'Run',
    date: '15/04/2018',
  },
  { 
    id: 1,
    timeSpent: '0:30',
    workoutType: 'Run',
    date: '15/04/2018',
  },
];

describe('<List/>', () => {
  it('Should component list items', () => {
    const wrapper = mount(<List items={mockList} />);

    expect(wrapper.find('tbody tr')).toHaveLength(mockList.length);
    expect(wrapper).toMatchSnapshot();
  });
});

