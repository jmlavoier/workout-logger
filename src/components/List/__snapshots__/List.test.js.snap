// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> Should component list items 1`] = `
<List
  changeOrder={[Function]}
  clickRemove={[Function]}
  items={
    Array [
      Object {
        "date": "15/04/2018",
        "id": 0,
        "timeSpent": "1:00",
        "workoutType": "Run",
      },
      Object {
        "date": "15/04/2018",
        "id": 1,
        "timeSpent": "0:30",
        "workoutType": "Run",
      },
    ]
  }
  orderBy={
    Object {
      "asc": true,
      "field": "teste",
    }
  }
>
  <table
    className="list"
  >
    <thead>
      <tr>
        <td>
          <OrderButton
            changeOrder={[Function]}
            fieldName="timeSpent"
            orderBy={
              Object {
                "asc": true,
                "field": "teste",
              }
            }
          >
            <div
              className="order-button"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="presentation"
            >
              Tempo
            </div>
          </OrderButton>
        </td>
        <td>
          <OrderButton
            changeOrder={[Function]}
            fieldName="workoutType"
            orderBy={
              Object {
                "asc": true,
                "field": "teste",
              }
            }
          >
            <div
              className="order-button"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="presentation"
            >
              Tipo
            </div>
          </OrderButton>
        </td>
        <td>
          <OrderButton
            changeOrder={[Function]}
            fieldName="date"
            orderBy={
              Object {
                "asc": true,
                "field": "teste",
              }
            }
          >
            <div
              className="order-button"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="presentation"
            >
              Data
            </div>
          </OrderButton>
        </td>
        <td>
          Â 
        </td>
      </tr>
    </thead>
    <tbody>
      <ItemList
        item={
          Object {
            "date": "15/04/2018",
            "id": 0,
            "timeSpent": "1:00",
            "workoutType": "Run",
          }
        }
        key="0"
        onClickRemove={[Function]}
      >
        <tr
          className="item-list"
        >
          <td>
            1:00
          </td>
          <td>
            Run
          </td>
          <td>
            20/04/2015
          </td>
          <td>
            <div
              onClick={[Function]}
              onKeyPress={[Function]}
              role="presentation"
            >
              <IconTrash>
                <svg
                  height="18"
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="
  M3 6v18h18v-18h-18zm5 14c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 
  1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 
  1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 
  1-1s1 .448 1 1v10zm4-18v2h-20v-2h5.711c.9 0 1.631-1.099 1.631-2h5.315c0 
  .901.73 2 1.631 2h5.712z
"
                    fill="#f1f1f1"
                  />
                </svg>
              </IconTrash>
            </div>
          </td>
        </tr>
      </ItemList>
      <ItemList
        item={
          Object {
            "date": "15/04/2018",
            "id": 1,
            "timeSpent": "0:30",
            "workoutType": "Run",
          }
        }
        key="1"
        onClickRemove={[Function]}
      >
        <tr
          className="item-list"
        >
          <td>
            0:30
          </td>
          <td>
            Run
          </td>
          <td>
            20/04/2015
          </td>
          <td>
            <div
              onClick={[Function]}
              onKeyPress={[Function]}
              role="presentation"
            >
              <IconTrash>
                <svg
                  height="18"
                  viewBox="0 0 24 24"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="
  M3 6v18h18v-18h-18zm5 14c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 
  1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 
  1-1s1 .448 1 1v10zm5 0c0 .552-.448 1-1 1s-1-.448-1-1v-10c0-.552.448-1 
  1-1s1 .448 1 1v10zm4-18v2h-20v-2h5.711c.9 0 1.631-1.099 1.631-2h5.315c0 
  .901.73 2 1.631 2h5.712z
"
                    fill="#f1f1f1"
                  />
                </svg>
              </IconTrash>
            </div>
          </td>
        </tr>
      </ItemList>
    </tbody>
  </table>
</List>
`;
